hathor/indexes/deps_index.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2021 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Iterator, List
+
+from hathor.transaction import BaseTransaction
+from hathor.transaction.transaction_metadata import ValidationState
+
+if TYPE_CHECKING:  # pragma: no cover
+    from hathor.transaction.storage import TransactionStorage
+
+
+INF_HEIGHT: int = 2**32 - 1
+
+
+def get_validation_state(tx_storage: 'TransactionStorage', tx: bytes) -> ValidationState:
```

Comment:
Should it be part of the `BaseTransaction`?

Reply:
This is somewhat similar to `BaseTransaction.get_requested_from_height`, although this _could_ be useful for other places (while `get_requested_from_height` is much less likely to be useful in other places) I'm still unsure, so I'm OK either way.

Reply:
I ended up removing this function. It made sense before when the index had uses from places where it had the hash of a transaction and needed the validation state. Now that it is refactored in a way in which that does not happen, it ends up being easier to just use `tx.get_metadata().validation` instead of `get_validation_state(tx_hash, tx_storage)` (because we had tx_hash and tx_storage separately).

Reply:
Ok. Thanks!